l10n_th_withholding_tax_cert/tests/test_wt_cert.py

```diff
@@ -156,26 +158,108 @@ def test_01_create_wt_cert_payment(self):
     def test_02_create_wt_cert_je(self):
         """ Journal Entry to WT Cert """
         misc_journal = self.browse_ref("account.miscellaneous_journal")
-        invoice_id = self._create_invoice(
-            False, misc_journal.id, "entry", self.wt_account
-        )
-        invoice_id.action_post()
+        invoice = self._create_invoice(False, misc_journal.id, "entry", self.wt_account)
+        self.assertEqual(invoice.state, "draft")
+        invoice.action_post()
+        self.assertEqual(invoice.state, "posted")
         # Create WT Cert from Journal Entry's Action Wizard
         ctx = {
-            "active_id": invoice_id.id,
-            "active_ids": [invoice_id.id],
+            "active_id": invoice.id,
+            "active_ids": [invoice.id],
             "active_model": "account.move",
         }
         res = self.wt_cert.with_context(ctx).action_create_withholding_tax_cert()
-        f = Form(self.env[res["res_model"]].with_context(res["context"]))
+        view = self.env["ir.ui.view"].browse(res["view_id"]).xml_id
+        f = Form(self.env[res["res_model"]].with_context(res["context"]), view=view)
         wizard = f.save()
-        wizard.write({"wt_account_ids": invoice_id.line_ids.mapped("account_id")})
+        wizard.write({"wt_account_ids": invoice.line_ids.mapped("account_id")})
         res = wizard.create_wt_cert()
         # New WT Cert
         ctx_cert = res.get("context")
-        ctx_cert.update({"default_income_tax_form": "pnd3", "wt_cert_income_type": "1"})
+        ctx_cert.update({"income_tax_form": "pnd3", "wt_cert_income_type": "1"})
         with Form(self.wt_cert.with_context(ctx_cert)) as f:
             f.income_tax_form = "pnd3"
         wt_cert = f.save()
         self.assertEqual(wt_cert.supplier_partner_id, self.partner_1)
-        invoice_id.button_wt_certs()
+        invoice.button_wt_certs()
+
+    def test_03_create_wt_cert_payment_multi(self):
+        """ Payments to WT Certs """
+        bank_journal = self.browse_ref("account.bank_journal")
+        expenses_journal = self.browse_ref("account.expenses_journal")
+        invoice = self._create_invoice(
+            self.partner_1.id, expenses_journal.id, "in_invoice"
+        )
+        invoice.action_post()
+        # Payment by writeoff with withholding tax account
+        ctx = {
+            "active_ids": [invoice.id],
+            "active_id": invoice.id,
+            "active_model": "account.move",
+        }
+        view_id = "account.view_account_payment_invoice_form"
+        with Form(self.account_payment.with_context(ctx), view=view_id) as f:
+            f.journal_id = bank_journal
+            f.amount = 97.0  # To withhold 3.0
+            f.payment_difference_handling = "reconcile"
+            f.writeoff_account_id = self.wt_account
+            f.writeoff_label = "Withhold 3%"
+        payment = f.save()
+        payment.post()
+        self.assertEqual(payment.state, "posted")
+        payment2 = payment.copy()
```

Comment:
When you copied the Payment, the WHT move line is not available.

![image](https://user-images.githubusercontent.com/1973598/98929031-275ad980-250d-11eb-9ee5-8aae56c33acd.png)

You should create 2 invoices -> 2 payments. If the code is long, try to shorten by doing subrouting.

And make sure you test the